gramps/gen/filters/rules/citation/_hasidof.py

```diff
@@ -46,3 +47,4 @@ class HasIdOf(HasGrampsId):
 
     name = _("Citation with <Id>")
     description = _("Matches a citation with a specified Gramps ID")
+    _get_raw_object_from_id_data = staticmethod(Database._get_raw_citation_from_id_data)
```

Comment:
The `_get_raw_citation_from_id_data` method is protected and should only be used in sub-classes.  It does not form part of the public API.

Reply:
Agreed. I also think it should be part of the public API. It makes no sense to construct the full object here when only the handle is required.